check_format.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const glob = require('glob');
+
+const CLANG_FORMAT = 'node_modules/.bin/clang-format';
+
+/**
+ * Checks that a path is formatted, fixing it if fix is true.
+ * @return false if the path was not formatted and we didn't fix it.
+ */
+function checkFile(path, fix) {
+  const source = fs.readFileSync(path, 'utf8');
+  const formatted = childProcess.execFileSync(CLANG_FORMAT, [path], {encoding: 'utf8'});
+  if (source !== formatted) {
+    if (!fix) return false;
+    fs.writeFileSync(path, formatted);
+  }
+  return true;
+}
+
+function main(args) {
+  const fix = args[0] === '--fix';
+  const sourceFiles = glob.sync('{*.js,src/**/*.[jt]s,test/**/*.[jt]s}');
+  for (const path of sourceFiles) {
+    if (!checkFile(path, fix)) {
```

Comment:
It's not clear that checkFile will overwrite the file if fix is true here.  I'd prefer if there was clear fix codepath and another clear check codepath.

Reply:
It's a bit complicated by how we need to read the text of the file in to fix a file, so the check and fix codepaths are very close to one another (see the 'fix' flag in checkFile).  Maybe I can rename something to address this?

Reply:
I'd be happy with a checkFile() and a fixFile() function that both called checkAndFixFile() with a different boolean if you're concerned about duplication.

Reply:
OK, PTAL